libp2p/muxers/mplex/lpchannel.nim

```diff
@@ -117,10 +118,10 @@ proc reset*(s: LPChannel) {.async.} =
         await s.conn.close()
         trace "Can't send reset message", s, conn = s.conn, msg = exc.msg
 
-    asyncSpawn resetMessage()
+    await resetMessage()
```

Comment:
the reset message is advisory - basically, it's a best-effort message that is sent in the hopes it will arrive and should therefore (in theory) not be allowed to block resetting the stream.

There's a bit of a terminology gap in libp2p as well - closing and resetting are used somewhat interchangeably. 

If we take `close` to mean what it does in BSD sockets, it's an operation that cannot fail / always succeeds - it can return an error but even if it does, the socket is considered closed afterwards.

In libp2p, `close` means more or less what `shutdown` means in bsd and `reset` is the "irrevokable close now" function - it's important that such a function doesn't get "stuck".

I'm not sure what the consequences of waiting here would be tbh, it deserves some thought and lots of testing.

Reply:
Could you please point out where it says it's a best-effort message? It doesn't seem like that here https://github.com/libp2p/specs/blob/master/mplex/README.md#resetting-a-stream.

Reply:
> This should generally only be used on error; 

the nature of the message is such that it is used when the stream is in an inconsistent state, ie when there is an error - because the stream is inconsistent, you also cannot guarantee that the message will be received by the other end and therefore you cannot rely on there being an answer to the message (it can get lost either locally, on the way to the other peer or the other peer may decide to not reply and if the other client is not reading the stream, it may get "blocked" forever).

Reply:
>  you cannot rely on there being an answer to the message

I don't think we expect any answer, do we?

Reply:
from the point of view of `await`, finishing the future is an answer - we don't expect an explicit message as answer, but from a buffer perspective, the future will not be finished until it has been given to the OS and the OS might not accept it until the other side sends an ACK confirming the receipt of earlier queued-up bytes.

Reply:
Ok, I reverted the change. I think it's unrelated to the PR anyway.